packages/cli/src/oclif/commands/canary/create.js

```diff
@@ -0,0 +1,97 @@
+const ZapierBaseCommand = require('../../ZapierBaseCommand');
+const { createCanary, listCanaries } = require('../../../utils/api');
+
+class CanaryCreateCommand extends ZapierBaseCommand {
+  async perform() {
+    const { versionFrom, versionTo, percent, duration } = this.args;
+
+    this.validateVersions(versionFrom, versionTo);
+    this.validatePercent(percent);
+    this.validateDuration(duration);
+
+    const activeCanaries = await listCanaries();
+    if (activeCanaries.objects.length > 0) {
+      const existingCanary = activeCanaries.objects[0];
+      const secondsRemaining = existingCanary.until_timestamp - Math.floor(Date.now() / 1000);
+      this.log(`A canary deployment already exists from version ${existingCanary.from_version} to version ${existingCanary.to_version}, there are ${secondsRemaining} seconds remaining.
```

Comment:
@standielpls one thing I'll call out here - since the API currently only allows one canary at a time, I've changed this code to block creation of a new one if _any_ canary exists. Initially it checked for a canary matching the `versionTo` and `versionFrom` args, but I think this is simpler and offers less potential for confusion

So tldr, only one canary can exist at a time

Reply:
I also don't super love the `activeCanaries.objects[0]` but the API returns the array right now, so 🤷